lib/rubocop/cop/lint/block_alignment.rb

```diff
@@ -64,24 +117,47 @@ def check_block_alignment(start_node, block_node)
           # blocks.
           match = /\S.*/.match(do_loc.source_line)
           indentation_of_do_line = match.begin(0)
-          return unless end_loc.column != indentation_of_do_line
-
-          add_offense(block_node,
-                      end_loc,
-                      format(MSG, end_loc.source, end_loc.line, end_loc.column,
-                             start_loc.source.lines.to_a.first.chomp,
-                             start_loc.line, start_loc.column,
-                             alt_start_msg(match, start_loc, do_loc,
-                                           indentation_of_do_line)))
+          return unless end_loc.column != indentation_of_do_line ||
+                        style == :start_of_line
+
+          {
+            source: match[0],
+            line: do_loc.line,
+            column: indentation_of_do_line
+          }
+        end
+
+        def loc_to_source_line_column(loc)
+          {
+            source: loc.source.lines.to_a.first.chomp,
+            line: loc.line,
+            column: loc.column
+          }
         end
 
-        def alt_start_msg(match, start_loc, do_loc, indentation_of_do_line)
-          if start_loc.line == do_loc.line &&
-             start_loc.column == indentation_of_do_line
+        def alt_start_msg(start_loc, source_line_column)
+          if style != :either
+            ''
+          elsif start_loc.line == source_line_column[:line] &&
+                start_loc.column == source_line_column[:column]
             ''
           else
-            " or `#{match[0]}` at #{do_loc.line}, #{indentation_of_do_line}"
+            ' or ' + format_source_line_column(source_line_column)
+          end
+        end
+
+        def format_source_line_column(source_line_column)
+          "`#{source_line_column[:source]}` at #{source_line_column[:line]}, " \
+          "#{source_line_column[:column]}"
+        end
+
+        def compute_start_col(ancestor_node, node)
+          if style == :start_of_block
+            do_loc = node.loc.begin
+            match = /\S.*/.match(do_loc.source_line)
```

Comment:
`=~` returns the start of the first match as an integer; with an adjustment to the regexp you could make this code a bit more concise that way. But actually, I think we already have a method in `Util` which could be used instead.